src/backend/mouse_cache.rs

```diff
@@ -9,6 +9,7 @@ use crate::event::MouseButtonState;
 pub struct MouseCache {
 	mouse_buttons: BTreeMap<DeviceId, MouseButtonState>,
 	mouse_position: BTreeMap<(WindowId, DeviceId), PhysicalPosition<f64>>,
+	cursor_direction: BTreeMap<(WindowId, DeviceId), PhysicalPosition<f64>>,
```

Comment:
This is -1.0, 0.0 or +1.0 now right? I think that's too specialized to store here. I do think it makes sense to add the previous mouse position to the cache (probably saving two positions in `mouse_position`). Then we can add `prev_position` to `MouseMove` events, which will be useful for user event handlers too.